tests/helpers.rs

```diff
@@ -31,132 +33,74 @@ fn get_random_tcp_port() -> u16 {
         .port()
 }
 
-#[derive(Debug)]
-pub struct CleanupGuard {
-    child: Option<Child>,
-
-    already_running: bool,
+pub struct TestEnv {
+    nats_server: ContainerAsync<GenericImage>,
+    wasmcloud_hosts: Vec<ContainerAsync<GenericImage>>,
 }
 
-impl Drop for CleanupGuard {
-    fn drop(&mut self) {
-        if !self.already_running {
-            match std::process::Command::new("wash").args(["down"]).output() {
-                Ok(o) if !o.status.success() => {
-                    eprintln!(
-                        "Error stopping wasmcloud host: {}",
-                        String::from_utf8_lossy(&o.stderr)
-                    )
-                }
-                Err(e) => eprintln!("Error stopping wasmcloud host: {e}"),
-                _ => (),
-            }
-        }
+impl TestEnv {
+    pub async fn nats_port(&self) -> u16 {
+        self.nats_server
+            .get_host_port_ipv4(DEFAULT_NATS_PORT)
+            .await
+            .unwrap()
     }
-}
-
-/// Configuration struct for wash instances that are used for testing
-#[derive(Debug, Default)]
-pub struct TestWashConfig {
-    /// Port on which to run wasmCloud
-    pub nats_port: Option<u16>,
 
-    /// Only connect to pre-existing NATS instance
-    pub nats_connect_only: bool,
-}
-
-impl TestWashConfig {
-    /// Build a test wash configuration with randomized ports
-    pub async fn random() -> Result<TestWashConfig> {
-        let nats_port = Some(get_random_tcp_port());
-
-        Ok(TestWashConfig {
-            nats_port,
-            ..TestWashConfig::default()
-        })
+    pub async fn nats_url(&self) -> String {
+        let nats_host = self.nats_server.get_host().await.unwrap();
```

Comment:
Same thing re: expect

Reply:
I ended up returning the error up to the caller so that it can be surfaced appropriately in the calling code